semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,456 @@
+    # define header
+    updatelog.columns = [
+        "obs_number",
+        "obs_key",
+        "obs_mean",
+        "obs_std",
+        "status",
+        "sim_mean",
+        "sim_std",
+    ]
+    # drop last line ('=====...====')
+    # drop last row i.e. end of file line
+    updatelog.drop(updatelog.tail(1).index, inplace=True)
+    # ---------------------------------
+    # add proper name for obs_key in rows where value is '...'
+    replace = "..."
+    for i in range(len(updatelog.index)):
+        obskey_val = updatelog.at[i, "obs_key"]
+        if obskey_val == replace:
+            updatelog.at[i, "obs_key"] = key
+
+    return updatelog
+
+
+def make_obs_groups(key_map):
+    """Create a mapping of observation groups, the names will be:
+    data_key -> [obs_keys] and All_obs-{missing_obs} -> [obs_keys]
+    and All_obs -> [all_obs_keys]
+    """
+    combinations = key_map.copy()
+    if len(combinations) == 1:
+        return combinations
+    # We want all observations and all observations
+    # minus individual observations:
+    for combination_length in (len(key_map.keys()) - 1, len(key_map.keys())):
+        if combination_length == 1:
+            # We have hit a case with two observation groups, now we only
+            # need individual group and all obs.
+            continue
+        for subset in itertools.combinations(key_map.keys(), combination_length):
+            obs_group = list(
+                itertools.chain.from_iterable([key_map[key] for key in subset])
+            )
+            if combination_length == len(key_map.keys()) - 1:
+                missing_obs = [x for x in key_map.keys() if x not in set(subset)]
+                assert len(missing_obs) == 1
+                name = f"All_obs-{missing_obs[0]}"
+            else:
+                name = "All_obs"
+            combinations[name.replace(":", "_")] = obs_group
+    return combinations
+
+
+def list_active_observations(df_update_log):
+
+    """To get the inactive vs total observation info."""
+    df_active = df_update_log[(df_update_log["status"] == "Active")]
+    return str(len(df_active)) + " active/" + str(len(df_update_log.index))
+
+
+def create_path(runpathdf, target_name):
+    """to create path for output data"""
+    return tempfile.mkdtemp()
+
+
+def list_observations_misfit(obs_keys, df_update_log, misfit_df):
+    """To get the misfit for total observations (active/inactive)."""
+    misfit_values = []
+    total_obs_nr = len(df_update_log[df_update_log.status.isin(["Active", "Inactive"])])
+    for key in obs_keys:
+        misfit_name = "MISFIT:" + key
+        misfit_values.append(misfit_df[misfit_name].mean())
+    return np.sum(misfit_values) / total_obs_nr
+
+
+def get_field_grid_char(input_grid):
+    """Get field grid characteristics/coordinates"""
+    input_grid = Path(input_grid).with_suffix("")
+    try:
+        mygrid = Grid(input_grid, fformat="eclipserun")
+        dataframeg = mygrid.dataframe(activeonly=False)
```

Comment:
I suggest renaming slightly. We can use `load` instead of `get` and perhaps not use `char` which is very much linked to `characters`, thus I would expect such a function to return a string or equivalent.

How about:
```
def load_grid_to_dataframe(filepath):
    fpath = Path(filepath).with_suffix("")
    try:
        grid = Grid(fpath, fformat="eclipserun")
        return grid.dataframe(activeonly=False)
    except...
```
?

Reply:
suggestion implemented